iceberg/iceberg-handler/src/test/queries/positive/iceberg_major_compaction_partition_evolution_w_id_spec_w_filter.q

```diff
@@ -1,76 +0,0 @@
--- SORT_QUERY_RESULTS
```

Comment:
I think this this test needs to be fixed instead of being deleted.
It tests compaction with a WHERE clause on a table with identity partition spec.
The test is failing because the predicate in the WHERE clause is not on partitioned columns.
It should be fixed by making the predicate on the partitioned columns.

Reply:
I deleted because there is a test for compaction with WHERE clause:
iceberg_major_compaction_partition_evolution_w_dyn_spec_w_filter.q

It would make no sense to have two with the same goal, and this one is no longer valid, since this PR does not allow anymore to have other columns in compaction where then partition columns. Modifying would lead to  make a copy of iceberg_major_compaction_partition_evolution_w_dyn_spec_w_filter.q .

Reply:
`iceberg_major_compaction_partition_evolution_w_dyn_spec_w_filter.q` is not duplicated test. It tests compaction with a WHERE clause on a table with dynamic partition spec whereas the one you deleted tests compaction with a WHERE clause on a table with identity partition spec. You can fix it by making the WHERE predicate use partition columns.